### PR TITLE
Remove gen_zero/gen_one

### PR DESCRIPTION
**Pull request by tautschnig**
As discussed in #400, `gen_zero` (and thus likely also `gen_one`) should be replaced by `from_integer` or other means of initialisation. This pull request removes all uses of `gen_zero` and `gen_one`.

### CODE REVIEW

**Comment by kroening** [comment]
f01581e appears to be unrelated?

**Comment by tautschnig** [comment]
I have removed the unrelated commit, and placed it in #451 instead.

File: src/util/expr_util.cpp [hunk 1]
```diff
@@ -321,8 +321,19 @@ exprt is_not_zero(
   irep_idt id=
     src_type.id()==ID_floatbv?ID_ieee_float_notequal:ID_notequal;
 
-  exprt zero=gen_zero(src_type);
-  assert(zero.is_not_nil());
+  exprt zero;
+  if(src_type.id()==ID_floatbv ||
+     src_type.id()==ID_fixedbv)
+  {
+    const floatbv_typet &type=to_floatbv_type(src_type);
```
Comment (peterschrammel) [inline src/util/expr_util.cpp]: This will abort for fixedbv.

Comment (peterschrammel) [inline src/util/expr_util.cpp]: Wouldn't it be cleaner to go through the ieee_floatt::make_zero().to_expr() route?

File: src/pointer-analysis/value_set_dereference.cpp [hunk 2]
```diff
@@ -967,7 +967,7 @@ bool value_set_dereferencet::memory_model_bytes(
     {
       binary_relation_exprt
         offset_lower_bound(offset, ID_lt,
-                           gen_zero(offset.type()));
+                           from_integer(0, offset.type()));
```
Comment (peterschrammel) [inline src/pointer-analysis/value_set_dereference.cpp]: cpplint

**Comment by tautschnig** [comment]
@peterschrammel Thanks a lot for the very good comments - indeed I have reworked the patch quite a bit for all floatbv/fixedbv related parts. Could you please take another look?

**Comment by kroening** [comment]
Looks ready to merge once rebased.

**Comment by tautschnig** [comment]
@kroening rebase done!

**Comment by tautschnig** [comment]
Note that testing is pointless/impossible until #467 is merged.

**Comment by tautschnig** [comment]
With 31ce11f this has actually become a bugfix.

**Pull request merged**